fdbserver/storageserver.actor.cpp

```diff
@@ -1521,7 +1521,14 @@ Future<Version> waitForVersion(StorageServer* data, Version commitVersion, Versi
 		return transaction_too_old();
 	} else {
 		if (commitVersion < data->oldestVersion.get()) {
-			return data->oldestVersion.get();
+			if (data->version.get() < readVersion) {
+				// Majority of the case, try using higher version to avoid
```

Comment:
It would be nice if you could extend this comment to say that it is correct to read at the oldest version (or at any version in the range [oldestVersion, data->version.get()] but we are not choosing to read at the oldest version for the reason that you mentioned. Thanks!

Reply:
Added more comments here.